models/feature-discovery-v1/model.jl

```diff
@@ -0,0 +1,126 @@
+using RxInfer, LinearAlgebra
+
+function create_feature_functions_1(x_dim)
+    linear = [(x) -> getindex(x, i) for i in 1:x_dim]
+    quadratic = [(x) -> getindex(x, i)^2 for i in 1:x_dim]
+    pairwise = [(x) -> getindex(x, i - 1) * getindex(x, i) for i in 2:x_dim]
+    tripplewise = [(x) -> getindex(x, i - 2) * getindex(x, i - 1) * getindex(x, i) for i in 3:x_dim]
+    return vcat(linear, quadratic, pairwise, tripplewise)
+end
+
+@model function feature_regression_unknown_noise(ϕs, x, y)
+    # Prior distribution over parameters ω
+    ω ~ Uninformative()
+
+    # Prior distribution over noise precision s
+    s ~ Uninformative()
+
+    if length(x) != length(y)
+        throw(DimensionMismatch("x and y must have the same length"))
+    end
+
+    M = length(x)
+
+    for i in 1:M
+        ϕx = [ϕ(x[i]) for ϕ in ϕs]
+        y[i] ~ Normal(mean = dot(ϕx, ω), precision = s)
+    end
+end
+
+@model function feature_regression_predictive(ϕs, x, y, priors)
+    ω ~ priors[:ω]
+    s ~ priors[:s]
+    ϕx = [ϕ(x) for ϕ in ϕs]
+    y ~ Normal(mean = dot(ϕx, ω), precision = s)
+end
+
+function initial_state(arguments)
+    return Dict{String, Any}("number_of_iterations" => arguments["number_of_iterations"])
+end
+
+function initial_parameters(arguments)
+    return Dict{String, Any}(
+        "omega_mean" => nothing, "omega_covariance" => nothing, "noise_shape" => nothing, "noise_scale" => nothing
+    )
+end
+
+function run_inference(state, parameters, data)
+    @debug "Running inference in FeatureDiscovery-v1 model" state parameters data
+
+    x = Float64.(data["x"])
+    phi_s = create_feature_functions_1(length(x))
+
+    omega_distribution = MvNormalMeanCovariance(parameters["omega_mean"], parameters["omega_covariance"])
+    noise_distribution = GammaShapeScale(parameters["noise_shape"], parameters["noise_scale"])
+
+    priors = Dict(:ω => omega_distribution, :s => noise_distribution)
+
+    # Create initialization for the inference
+    init = @initialization begin
+        μ(ω) = MvNormalMeanCovariance(parameters["omega_mean"], parameters["omega_covariance"])
+        q(s) = GammaShapeScale(parameters["noise_shape"], parameters["noise_scale"])
+    end
+
+    inference_results = infer(
+        model = feature_regression_predictive(ϕs = phi_s, priors = priors),
+        data = (x = x, y = missing),
+        predictvars = (y = KeepLast(),),
+        initialization = init,
+        iterations = state["number_of_iterations"],
+        constraints = MeanField()
```

Comment:
I can't remember for sure, but when we have a clamped missing and MeanField() constraint, would it eventually enforce structured on between `dot(ϕx, ω)` and `y`? Could it be that this is why the variance over the prediction was very small?